gqlalchemy/memgraph.py

```diff
@@ -543,3 +544,44 @@ def create_relationship(self, relationship: Relationship) -> Optional[Relationsh
         )
 
         return self.get_variable_assume_one(results, "relationship")
+
+
+class IntegratedAlgorithm(ABC):
+    """Abstract class modeling in-Memgraph graph algorithms.
+
+    These algorithms are integrated into Memgraph codebase and are called
+    within an edge part of a query. For instance:
+    MATCH p = (:City {name: "Paris"})
+          -[:Road * bfs (e, v | e.length <= 200 AND v.name != "Metz")]->
+          (:City {name: "Berlin"})
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def __str__(self) -> str:
+        """Instance of IntegratedAlgorithm extended object is used as a string.
+
+        Raises:
+            NotImplementedError: Inheriting class did not define its string
+            representation
+        """
+        raise NotImplementedError("Algorithm should define its str representation")
+
+    @staticmethod
+    def get_lambda(expression: str) -> str:
```

Comment:
Would it be better to call it `to_cypher_lambda()`, since we already have `to_cypher_value()`, `to_cypher_properties()`, etc.?

Reply:
Sure!